Fix/dlc v3 fixes

pascalhuerten opened the pull request:
Beim DLC sind uns noch ein paar Fehler aufgefallen in Verbindung mit den neuen v3.x Versionen des Booking-Backends.
Ich habe versucht Fixes dazu in unserer Fork vorzunehmen, um das Backend in Verbindung mit unserem DLC zum Laufen zu bringen. Könntet ihr diese Fixes einmal gegenprüfen und gegebenenfalls in eure Entwicklung mitaufnehmen? Im folgenden beschreibe ich die einzelnen Fixes und die Probleme, die diese beheben sollen. Manche der Lösungnen sind vielleicht nicht vollständig oder ihr habt noch andere Ideen wie diese sinnvoller zu lösen wären.

-------

### SSO User Role Management Enhancements:
* [`src/commons/data-managers/tenant-manager.js`](diffhunk://#diff-3fec373c18908402f22757aad01209c525c4024dee56bcd14f2ee83e837df334R118-R132): Enhanced the `addUserRole` method to handle cases where the user or users array does not yet exist in the tenant. This will make sure that when a user signs up using sso, the user will get the appropriate user roles according to the role mapping with the keycloak application.

### Null Safety Improvements regarding PriceCategory:
* [`src/commons/services/checkout/item-checkout-service.js`](diffhunk://#diff-db6125299a1ce5eadfe2664471eb4220f214f8a77fd52aadf9b4bd111e5e4f7cL89-R91): Added optional chaining (`?.`) to handle cases where `getPriceCategory` might return null in multiple methods. [[1]](diffhunk://#diff-db6125299a1ce5eadfe2664471eb4220f214f8a77fd52aadf9b4bd111e5e4f7cL89-R91) [[2]](diffhunk://#diff-db6125299a1ce5eadfe2664471eb4220f214f8a77fd52aadf9b4bd111e5e4f7cL173-R173) [[3]](diffhunk://#diff-db6125299a1ce5eadfe2664471eb4220f214f8a77fd52aadf9b4bd111e5e4f7cL188-R188)
* There could be more places where the same issues might occur. I only tested the features, that we (DLC) currently use.

### Child Bookable Availability Check Fix:
* [`src/commons/services/checkout/item-checkout-service.js`](diffhunk://#diff-db6125299a1ce5eadfe2664471eb4220f214f8a77fd52aadf9b4bd111e5e4f7cL367-R367): Corrected a logical error in the availability check of child bookables. A child bookable will now be unavailable when the variable `isAvailable` is false.

### Fix logical condition in updateBookable and updatEvent methods
* [`src/platform/api/controllers/bookable-controller.js`](diffhunk://#diff-45454fa6514900542579a2e660dd42f77212dd65f941983ea2bdf908399b25b2L321-R324): Fixed the condition to properly handle cases where `existingBookable` might be null. This was required, since we are setting a desired id in the request, when creating new bookables using the api. This lead to the booking backend to choose to use updateBookable instead of createBookable, in which case existingBookbale would be null.
* [`src/platform/api/controllers/event-controller.js`](diffhunk://#diff-baae8cab658fed8ed2de175f7d7f344f3cb3a7636ca4e37bee68185eeda87a65L123-R123): Fixed the condition to properly handle cases where `existingEvents` might be null. Same situation as above but for creating new events using updateEvent instaead of createEvent.

### User Permission Check Fix for getting assigned bookings:
* [`src/platform/api/controllers/booking-controller.js`](diffhunk://#diff-04953590b5890767ede735fa2b8c37e20922ededc8a28184a361337f84ec2374L120-R125): Replaced a placeholder permission check with a call to `UserManager.hasPermission` to verify user permissions properly for getting user assinged bookings. Before this method was still trying to check the tenant property of a user, wich is not part of the user model since v3 anymore.

pascalhuerten committed:
On adding a user role also add the user to the tenant if it does not already exist

pascalhuerten committed:
Account for the possibility that getPriceCategory can return null

pascalhuerten committed:
Fix availability check for child bookings

pascalhuerten committed:
Allow existing bookables or events to be non existent, when calling updateBookablel or updateEvent

pascalhuerten committed:
Fix user permission ckeck when getting assigned bookings

kpw89 committed:
fixed issue where calling route /bookables/:id/bookings returned array with duplicate entries. ( getBookableBookings() in hub )

kpw89 commented:
Beim Aufruf der Route /bookables/:id/bookings wurden alle Buchungen im zurückgegebenen Array doppelt aufgeführt.

### Fix : route /bookables/:id/bookings returned bookings with duplicates 
**_src/platform/api/controllers/booking-controller.js:
In getRelatedBookings()_**

- Moved bookings.concat() out of for-loop to prevent duplicates (in relatedBookings and ParentBookings)

- Added a safeguard to prevent concating empty bookings when parentBookings/relatedBookings is undefined or null.

pascalhuerten committed:
Fix issue trying to remove a workflow task that does not exist during booking cancelation if the tenant has no workflows assigned

pascalhuerten commented:
Beim Stornieren von Buchungen ist ein Fehler aufgetreten, wenn keine Workflows existieren. Der Fall dass keine Workflow Tasks in der Datenbank gefunden wird, wurde nicht korrekt abgefangen, da die Methode WorkflowManager.getWorkflow eine leere Map {} in diesem Fall zurückgibt. WorkflowService.removeTask prüfte allerdings bisher nur ob das Ergebnis True oder false entspricht. Stattdessen wird nun geprüft ob das Ergebnis auch die benötigte Eigenschaft "states" enthält.

Zu behebender Fehler: 
```
{
   "message":"Cannot read properties of undefined (reading 'find')",
   "name":"TypeError",
   "stack":"TypeError: Cannot read properties of undefined (reading 'find') at removeTask (/app/src/commons/services/workflow/workflow-service.js:260:35) at WorkflowService.removeTask (/app/src/commons/services/workflow/workflow-service.js:88:5) at process.processTicksAndRejections (node:internal/process/task_queues:95:5) at async removeBooking (/app/src/platform/api/controllers/booking-controller.js:424:11)"
}
```

Fix:
src/commons/services/workflow/workflow-service.js:82 in removeTask()
~`if (!workflow) { ...`~
`if (!workflow || !workflow.states) { ...`

Marvin-Anders committed:
Update bookable-controller.js

Use optional chaining for public‐bookable check

Simplify the transition logic by replacing the explicit null check with optional chaining.

Marvin-Anders committed:
Remove permission check from getAssignedBookings

Since the User model no longer contains a tenant field and 
getAssignedBookings inherently filters by assignedUserId and tenantId, 
drop the explicit permission validation. This simplifies the flow 
and relies on the query to enforce that users only retrieve their 
own bookings within the specified tenant.

Marvin-Anders committed:
Use optional chaining for public‐event check

Simplify the transition logic by replacing the explicit null check with optional chaining.

Marvin-Anders committed:
Update tenant-manager.js

refactor: simplify addUserRole by delegating user-creation logic to service layer
